core/src/edu/asu/c3simulator/util/Product.java

```diff
@@ -44,7 +47,7 @@ private Image getImage(String imagePath)
 	
 	public Image getProductImage()
 	{
-		return productImage;
+		return getImage(imagePath);
```

Comment:
Yes, we do need unique copies of the image. A lot of screens use the product image as icons and add listeners to them. Having new images returned will make it possible to add different listeners to the same product's icon. For example, screens that manage current product and the marketing screen should all get the same list pf product objects from the company data layer. So they should each create a new icon for user interaction appropriate to their respected screen.